packages/backend-defaults/src/entrypoints/database/connectors/postgres.ts

```diff
@@ -121,6 +129,41 @@ export async function buildPgDatabaseConfig(
   };
 }
 
+export async function buildAzurePgConfig(config: any, sanitizedConfig: any) {
+  const { DefaultAzureCredential } = require('@azure/identity');
+
+  // By default get a new token starting three minutes before the old one expires
+  const allowedClockSkewMs =
+    config.connection.allowedClockSkewMs !== undefined
+      ? config.connection.allowedClockSkewMs
+      : 180000;
+
+  async function getConnectionConfig() {
+    const credential = new DefaultAzureCredential();
+    const token = await credential.getToken(
+      'https://ossrdbms-aad.database.windows.net',
+    );
+
+    const connectionConfig = {
+      ...sanitizedConfig.connection,
+      password: token.token,
+      expirationChecker: () => {
+        /* return true if the token is near to or past its expiration. */
+        const isExpired =
+          token.expiresOnTimestamp - allowedClockSkewMs <= Date.now();
```

Comment:
The default refresh offset in The Microsoft authentication library is [5 minutes](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/d28085905644510fbcfacfed98ed1b43014b67c1/lib/msal-common/src/client/RefreshTokenClient.ts#L54). Maybe you should use 5 minutes as a constant here and then subtract the `allowedClockSkewMs` as an additional offset (on top of the default 5 minutes).
```suggestion
          token.expiresOnTimestamp - fiveMinutes - allowedClockSkewMs <= Date.now();
```
or rename the `allowedClockSkewMs` (configuration) variable to something like `tokenExpirationOffset`
```suggestion
          token.expiresOnTimestamp - tokenExpirationOffset <= Date.now();
```

Reply:
Thanks for the reference to the MSAL library. I see its configuration has an optional property named `tokenRenewalOffsetSeconds` which defaults, as you said, to 5 minutes. What if I adopted that name and its default value for our configuration? I don't think we need _both_ that offset as well as a clock skew offset.

Reply:
That makes sense. I feel it's also bit more descriptive and applicable than `allowedClockSkewMs`. Normally the offset is also meant to have a chance to recover for failures when renewing the token.